includes/compat/trait-wc-stripe-deposits.php

```diff
@@ -0,0 +1,457 @@
+<?php
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Trait for Deposits compatibility.
+ *
+ * @since x.x.x
+ */
+trait WC_Stripe_Deposits_Trait {
+
+	/**
+	 * Stores a flag to indicate if the deposits integration hooks have been attached.
+	 *
+	 * The callbacks attached as part of maybe_init_deposits() only need to be attached once to avoid duplication.
+	 *
+	 * @var bool False by default, true once the callbacks have been attached.
+	 */
+	private static $has_attached_deposits_integration_hooks = false;
+
+	/**
+	 * Initialize deposits hooks.
+	 *
+	 * @since x.x.x
+	 */
+	public function maybe_init_deposits() {
+		if ( ! $this->is_deposits_enabled() ) {
+			return;
+		}
+
+		$this->supports[] = 'forced-tokenization'; // @phpstan-ignore-line (supports is defined in the classes that use this trait)
+
+		add_action( 'wc_deposits_' . $this->id . '_charge_order_token', [ $this, 'charge_order_token' ], 10, 2 ); // @phpstan-ignore-line (id is defined in the classes that use this trait)
+
+		/**
+		 * The callbacks attached below only need to be attached once. We don't need each gateway instance to have its own callback.
+		 * Therefore we only attach them once on the main `stripe` gateway and store a flag to indicate that they have been attached.
+		 */
+		if ( self::$has_attached_deposits_integration_hooks || WC_Gateway_Stripe::ID !== $this->id ) { // @phpstan-ignore-line (id is defined in the classes that use this trait)
+			return;
+		}
+
+		add_filter( 'pre_wc_deposits_get_order_payment_token', [ $this, 'get_order_payment_token' ], 10, 2 );
+
+		add_action( 'wc_deposits_delete_order_payment_token', [ $this, 'delete_order_payment_token' ], 10, 2 );
+
+		self::$has_attached_deposits_integration_hooks = true;
+	}
+
+	/**
+	 * Filter the order payment token to include the source and intent.
+	 *
+	 * Runs on the `pre_wc_deposits_get_order_payment_token` filter.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array $token The order payment token.
+	 * @param WC_Order $top_most_order The top most order in the chain.
+	 * @return array The modified order payment token.
+	 */
+	public function get_order_payment_token( $token, $top_most_order ) {
+		if ( $top_most_order->get_payment_method() !== $this->id ) {
+			return $token;
+		}
+
+		$intent = $this->get_intent_from_order( $top_most_order );
+
+		if (
+			! $intent ||
+			! isset( $intent->setup_future_usage ) ||
+			! 'off_session' === $intent->setup_future_usage
```

Comment:
`! 'off_session' === …` parses as `(! 'off_session') === …`, which is always `false === string`

```suggestion
			! ( 'off_session' === $intent->setup_future_usage )
```